app/src/main/java/bacon/john/bitcoinpricewatch/CurrencyListActivity.java

```diff
@@ -0,0 +1,89 @@
+package bacon.john.bitcoinpricewatch;
+
+import android.content.Intent;
+import android.os.Bundle;
+import android.support.v4.app.FragmentActivity;
+
+import bacon.john.bitcoinpricewatch.bitcoin.BitcoinCurrencyManager;
+import bacon.john.bitcoinpricewatch.bitcoin.BitcoinCurrencyModel;
+import bacon.john.bitcoinpricewatch.localstorage.LocalStorageService;
+
+import java.util.List;
+
+/**
+ * An activity representing a list of Currencies. This activity
+ * has different presentations for handset and tablet-size devices. On
+ * handsets, the activity presents a list of items, which when touched,
+ * lead to a {@link CurrencyDetailActivity} representing
+ * item details. On tablets, the activity presents the list of items and
+ * item details side-by-side using two vertical panes.
+ * <p/>
+ * The activity makes heavy use of fragments. The list of items is a
+ * {@link CurrencyListFragment} and the item details
+ * (if present) is a {@link CurrencyDetailFragment}.
+ * <p/>
+ * This activity also implements the required
+ * {@link CurrencyListFragment.Callbacks} interface
+ * to listen for item selections.
+ */
+public class CurrencyListActivity extends FragmentActivity
+        implements CurrencyListFragment.Callbacks {
+
+    /**
+     * Whether or not the activity is in two-pane mode, i.e. running on a tablet
+     * device.
+     */
+    private boolean mTwoPane;
+
+    @Override
+    protected void onCreate(Bundle savedInstanceState) {
+        super.onCreate(savedInstanceState);
+
+        LocalStorageService.sharedInstance().initialize(getApplicationContext());
+
+        setContentView(R.layout.activity_currency_list);
+
+        if (findViewById(R.id.currency_detail_container) != null) {
+            // The detail container view will be present only in the
+            // large-screen layouts (res/values-large and
+            // res/values-sw600dp). If this view is present, then the
+            // activity should be in two-pane mode.
+            mTwoPane = true;
```

Comment:
Admittedly, leveraged a lot of boilerplate code Android Studio provided for me when I picked a master-detail UI. I understand that this works properly because of the above comment.